robottelo/cli/factory.py

```diff
@@ -83,23 +84,24 @@ def create_object(cli_object, options, values):
 
     """
     update_dictionary(options, values)
-    result = cli_object.create(options)
-
-    # If the object is not created, raise exception, stop the show.
-    if result.return_code != 0:
+    try:
+        result = cli_object.create(options)
+    except CLIReturnCodeError as err:
+        # If the object is not created, raise exception, stop the show.
         raise CLIFactoryError(
-            'Failed to create {0} with data:\n{1}'.format(
+            'Failed to create {0} with data:\n{1}\n{2}'.format(
                 cli_object.__name__,
-                json.dumps(options, indent=2, sort_keys=True)
+                json.dumps(options, indent=2, sort_keys=True),
+                err.msg,
```

Comment:
Thanks for adding this error message to the raised exception. I'm sure it'll aid in debugging when failures happen.